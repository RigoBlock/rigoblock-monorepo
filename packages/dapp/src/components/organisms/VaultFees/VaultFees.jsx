import './VaultFees.scss'
import BigNumber from 'bignumber.js'
import Button from '../../atoms/Button'
import PropTypes from 'prop-types'
import React from 'react'
import Title from '../../atoms/Title'

const VaultFees = ({ vault }) => {
  const transactionFee = vault.transactionFee.toFormat(2, BigNumber.ROUND_FLOOR)
  return (
    <div className="vault-fees">
      <Title>Fees</Title>
      <div className="fee-info">
        <span>Management</span>
        <span>{`0.00%`}</span>
      </div>
      <div className="fee-info">
        <span>Entry</span>
        <span>{`${transactionFee}%`}</span>
      </div>
      <div className="fee-info">
        <span>Exit</span>
        <span>{`${transactionFee}%`}</span>
      </div>
      {/* TODO: add action to open modal */}
      <Button onClick={() => {}}>Edit Fees</Button>
    </div>
  )
}

VaultFees.propTypes = {
  vault: PropTypes.object.isRequired
}

export default VaultFees
