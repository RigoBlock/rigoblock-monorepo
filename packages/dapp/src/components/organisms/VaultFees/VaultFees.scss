@import "../../_settings/variables.scss";

.vault-fees {
  .title {
    padding-bottom: $dist-2xs;
  }
  .fee-info {
    display: flex;
    justify-content: space-between;
    width: 160px;
    padding-bottom: $dist-xs;
  }
}
