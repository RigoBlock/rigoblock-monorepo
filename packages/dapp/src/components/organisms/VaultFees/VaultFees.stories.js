import '../../_settings/_base.scss'
import { BigNumber } from 'bignumber.js'
import { INVESTOR, MANAGER } from '../../../constants/user'
import { MemoryRouter } from 'react-router-dom'
import { select, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import React from 'react'
import VaultFees from './VaultFees'

const userTypes = {
  Manager: MANAGER,
  Investor: INVESTOR
}

const vault = {
  id: new BigNumber('0'),
  name: 'First Vault',
  symbol: 'asd',
  owner: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196',
  group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
  totalSupply: new BigNumber('28640000000000000000'),
  transactionFee: new BigNumber('2.50')
}

storiesOf('Organisms/VaultFees', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div style={{ width: '478px', background: 'white' }}>{story()}</div>
  ))
  .addDecorator(story => <MemoryRouter>{story()}</MemoryRouter>)
  .add('default', () => (
    <VaultFees
      vault={vault}
      userType={select('User type', userTypes, userTypes.Investor)}
    />
  ))
