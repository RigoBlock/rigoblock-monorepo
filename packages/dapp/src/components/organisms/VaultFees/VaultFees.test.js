import { BigNumber } from 'bignumber.js'
import { shallow } from 'enzyme'
import React from 'react'
import VaultFees from './VaultFees.jsx'
import toJson from 'enzyme-to-json'

const vault = {
  id: new BigNumber('0'),
  name: 'First Vault',
  symbol: 'asd',
  owner: '0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196',
  group: '0x7ce6e371085cb611fb46d5065397223ef2f952ff',
  totalSupply: new BigNumber('28640000000000000000'),
  transactionFee: new BigNumber('2.50')
}

describe('VaultFees component', () => {
  it('renders correctly', () => {
    expect(toJson(shallow(<VaultFees vault={vault} />))).toMatchSnapshot()
  })
})
