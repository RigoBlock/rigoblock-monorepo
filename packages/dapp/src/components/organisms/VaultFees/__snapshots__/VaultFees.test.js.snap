// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultFees component renders correctly 1`] = `
<div
  className="vault-fees"
>
  <Title
    className={null}
    size="medium"
  >
    Fees
  </Title>
  <div
    className="fee-info"
  >
    <span>
      Management
    </span>
    <span>
      0.00%
    </span>
  </div>
  <div
    className="fee-info"
  >
    <span>
      Entry
    </span>
    <span>
      2.50%
    </span>
  </div>
  <div
    className="fee-info"
  >
    <span>
      Exit
    </span>
    <span>
      2.50%
    </span>
  </div>
  <Button
    appearance="primary"
    onClick={[Function]}
    type="button"
  >
    Edit Fees
  </Button>
</div>
`;
